kernel/mmanutils.c

```diff
@@ -1155,6 +1152,37 @@ ssize_t myst_mman_pids_test(const void* addr, size_t length, pid_t pid)
     return _handle_mman_pids_op(MMAN_PIDS_OP_TEST, addr, length, pid);
 }
 
+bool myst_is_bad_addr(const void* addr, size_t length, int prot)
+{
+    bool ret = true;
+    uint64_t page_addr;
+
+    if (!addr || !length)
+        goto done;
+
+    /* check if the pages within the address range are unmapped (i.e.,
+     * associated pid is zero). */
+    page_addr = myst_round_down_to_page_size((uint64_t)addr);
+    if (myst_mman_pids_test((const void*)page_addr, length, 0) > 0)
```

Comment:
since addr is being rounded down, length needs to be updated.
[page_addr, page_addr+length] may not always enclose [addr, addr+length]

Reply:
the length will be rounded up in both `myst_mman_pids_test` and `myst_mman_get_prot`. But I can also do here to ensure the consistency.